src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/BlazorDebug/BlazorDebugConfigurationProvider.ts

```diff
@@ -23,17 +25,27 @@ export class BlazorDebugConfigurationProvider implements vscode.DebugConfigurati
             await this.launchApp(folder, configuration);
         }
 
-        const result = await vscode.commands.executeCommand<{
-            url: string,
-            inspectUri: string,
-            debuggingPort: number,
-        }>('blazorwasm-companion.launchDebugProxy', folder, configuration);
+        let inspectUri = '{wsProtocol}://{url.hostname}:{url.port}/_framework/debug/ws-proxy?browser={browserInspectUri}';
+        let url = 'https://localhost:5001';
+        try {
+            if (folder !== undefined) {
+                let folderPath = configuration.cwd ? configuration.cwd : fileURLToPath(folder.uri.toString());
+                folderPath = folderPath.replace('${workspaceFolder}', fileURLToPath(folder.uri.toString()));
+                const launchSettings = JSON.parse(readFileSync(join(folderPath, 'Properties', 'launchSettings.json'), 'utf8'));
+                if (launchSettings?.profiles && launchSettings?.profiles[Object.keys(launchSettings.profiles)[0]]?.inspectUri) {
+                    inspectUri = launchSettings.profiles[Object.keys(launchSettings.profiles)[0]].inspectUri;
+                    url = launchSettings.profiles[Object.keys(launchSettings.profiles)[0]].applicationUrl.split(';', 1)[0];
+                }
+            }
+        } catch (error: any) {
+            this.logger.logError('[DEBUGGER] Error while getting information from launchSettings.json: ', error as Error);
+        }
 
         await this.launchBrowser(
             folder,
             configuration,
-            result ? result.inspectUri : undefined,
-            result ? result.debuggingPort : undefined);
+            inspectUri,
+            url);
```

Comment:
Before, this was the result.debuggingPort, which seems like it would be a number.  Now it's a URL; is this expected?

Reply:
result.debuggingPort is not used anymore, this is expected.